### PR TITLE
install: add ipykernel as an optional requirement

### PR DESCRIPTION
**Pull request by superlopuh**
Also make the README instructions a bit clearer, I think it's nice to make clear that `-e` stands for `--editable`, meaning that the code will be updated as you update the repo.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,8 +50,8 @@ To contribute to the development of xDSL follow the subsequent steps.
 
 ```bash
 git clone https://github.com/xdslproject/xdsl.git
-pip install -e .
-pip install -e .[extras]
+pip install --editable .
+pip install --requirement requirements-optional.txt
```
Comment (georgebisbas) [inline README.md:54]: This second line now only installs optional and not the basic reqs

**Review by georgebisbas — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (superlopuh) [reply README.md:54]: I just checked, the first line installs the requirements for me.

**Review by webmiche — APPROVED** [review]
I think this needs a rebase. Afterward, I am fine with merging.

**Comment by webmiche** [comment]
Is there something happening here? @georgebisbas @superlopuh

**Review by georgebisbas — APPROVED** [review]
We can merge, yes

**Pull request merged**